toolsuite/src/main/java/com/faforever/neroxis/toolsuite/MapEnvTextureExporter.java

```diff
@@ -0,0 +1,69 @@
+package com.faforever.neroxis.toolsuite;
+
+import com.faforever.neroxis.cli.RequiredMapPathMixin;
+import com.faforever.neroxis.cli.VersionProvider;
+import com.faforever.neroxis.exporter.SCMapExporter;
+import com.faforever.neroxis.importer.MapImporter;
+import com.faforever.neroxis.map.SCMap;
+import com.faforever.neroxis.map.Symmetry;
+import com.faforever.neroxis.map.SymmetrySettings;
+import com.faforever.neroxis.mask.BooleanMask;
+import com.faforever.neroxis.mask.FloatMask;
+import com.faforever.neroxis.mask.NormalMask;
+import com.faforever.neroxis.util.ImageUtil;
+import picocli.CommandLine;
+
+import java.util.concurrent.Callable;
+
+@CommandLine.Command(name = "export-env-map", mixinStandardHelpOptions = true,
+        description = "Export the mapwide normal, waterDepth and shadow texture",
+        versionProvider = VersionProvider.class, usageHelpAutoWidth = true)
+public class MapEnvTextureExporter implements Callable<Integer> {
+    @CommandLine.Spec
+    private CommandLine.Model.CommandSpec spec;
+    @CommandLine.Mixin
+    private RequiredMapPathMixin requiredMapPathMixin;
+
+    @Override
+    public Integer call() throws Exception {
+        generateEnvTexture();
+        return 0;
+    }
+    
+    public void generateEnvTexture() throws Exception {
+        System.out.print("Generating env texture\n");
+        SCMap map = MapImporter.importMap(requiredMapPathMixin.getMapPath());
+
+        // Copied from TextureGenerator and BasicTextureGenerator
+        FloatMask heightMapSize = new FloatMask(map.getHeightmap(), (long) 0, new SymmetrySettings(Symmetry.NONE))
+                .startVisualDebugger()
+                .resample(map.getSize())
+                .divide(128f); // No idea why this is necessary
```

Comment:
I am not sure why you are doing this either. Normally it has to do with the scaling of the mask values but I am not sure what they are supposed to be here

Reply:
The mask values need to be the actual height of the heightmap, so around 0 to 50. This is needed so the normal map has the right strenght and the calculations for the water depth produce the correct results.
For some reason the heightmap values are initally to big by a factor of exactly 128.

Reply:
The 128 is because in the scmap binary the scale factor is 128. And if it is just about the normal map you could use the scalar argument to copy as normal mask

Reply:
It's also relevant for the water depth